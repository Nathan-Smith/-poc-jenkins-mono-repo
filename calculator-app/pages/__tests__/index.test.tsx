import React from 'react'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Home from '../index'

describe('index', () => {
  test('renders', () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<Home />)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })

  test('clicking 2 sets result to 2', () => {
    const { getByTestId } = render(<Home />)

    fireEvent.click(getByTestId(/keypad-2/i))

    expect(getByTestId(/result/i)).toHaveTextContent('2')
  })
})
