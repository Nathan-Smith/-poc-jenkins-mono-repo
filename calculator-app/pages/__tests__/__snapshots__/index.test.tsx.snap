// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders 1`] = `
<main
  className="main"
>
  <label
    className="mdc-text-field mdc-text-field--filled mdc-text-field--end-aligned result"
    style={
      Object {
        "gridArea": "result",
      }
    }
  >
    <span
      className="mdc-text-field__ripple"
    />
    <input
      className="mdc-text-field__input resultInput"
      data-testid="result-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <span
      className="mdc-line-ripple"
    />
  </label>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-1",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      1
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-2",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      2
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-3",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      3
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-4",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      4
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-5",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      5
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-6",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      6
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-7",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      7
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-8",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      8
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-9",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      9
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-add",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      +
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-0",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      0
    </span>
  </button>
  <button
    className="mdc-button mdc-button--outlined keypadButton"
    style={
      Object {
        "gridArea": "keypad-decimal",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      .
    </span>
  </button>
  <button
    className="mdc-button mdc-button--raised keypadButton keypadEqual"
    style={
      Object {
        "gridArea": "keypad-equal",
        "overflow": "hidden",
      }
    }
  >
    <span
      className="mdc-button__ripple"
    />
    <span
      className="mdc-button__label"
    >
      =
    </span>
  </button>
</main>
`;
